src/hidpp20.c

```diff
@@ -2179,6 +2128,29 @@ int hidpp20_onboard_profiles_write(struct hidpp20_device *device,
 	return rc < 0 ? rc : 0;
 }
 
+int
+hidpp20_onboard_profiles_commit(struct hidpp20_device *device,
+				struct hidpp20_profiles *profiles_list)
+{
+	struct hidpp20_profile *profile;
+	unsigned int i;
+	int rc;
+
+	for (i = 0; i < profiles_list->num_profiles; i++) {
+		profile = &profiles_list->profiles[i];
+
+		if (profile->enabled) {
+			rc = hidpp20_onboard_profiles_write_profile(device,
+								    profiles_list,
+								    i);
+			if (rc < 0)
+				return rc;
+		}
+	}
+
+	return hidpp20_onboard_profiles_write_dict(device, profiles_list);
```

Comment:
hmm, I wonder: what happens on partial failures here? either when two profiles write correctly and one doesn't, or when the profiles write but the dict fails?

Reply:
good question. Depending on where the error happens and what error is triggered, we might just have either part of the profiles working or the device in the factory mode (if the crcs are wrong).

I guess we should resync the state of the mouse at that point. Probably in a later commit.